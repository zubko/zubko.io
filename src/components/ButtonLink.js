import React from 'react';

import Link from './Link';

export default ({
  backgroundColor,
  borderColor,
  title,
  to,
  children,
  ...otherProps
}) => (
  <Link
    usePlainStyle
    to={to}
    css={[styles.button, { backgroundColor, '--boxShadowColor': borderColor }]}
    {...otherProps}>
    {children}
  </Link>
);

const styles = {
  button: {
    userSelect: 'none',
    padding: '0.3em 0.7em 0.2em',
    color: 'white',
    textDecoration: 'none',
    backgroundImage: 'none',
    textShadow: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: '6px',
    textAlign: 'center',
    verticalAlign: 'center',
    transform: 'translate(0, 0)',
    transition: 'all .2s ease',
    cursor: 'pointer',
    overflow: 'hidden',
    boxShadow: '0px 6px 0px var(--boxShadowColor, #000)',
    '&:hover': {
      color: 'white',
      transform: 'translate(0, -2px)',
      'box-shadow': '0px 8px 0px  var(--boxShadowColor, #000)',
      filter: 'brightness(110%)',
    },
    '&:active': {
      transform: 'translate(0, 3px)',
      'box-shadow': '0px 2px 0px  var(--boxShadowColor, #000)',
    },
  },
};
