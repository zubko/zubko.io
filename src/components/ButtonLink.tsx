/**
 * Component which renders a button to be used as a link
 */

import * as React from "react";
import Link from "./Link";

type Props = {
  backgroundColor: string;
  borderColor: string;
  to: string;
  children?: React.Node;
};

export default ({
  backgroundColor,
  borderColor,
  to,
  children,
  ...otherProps
}: Props) => (
  <Link
    usePlainStyle
    to={to}
    css={[styles.button, { backgroundColor, "--boxShadowColor": borderColor }]}
    {...otherProps}
  >
    {children}
  </Link>
);

const styles = {
  button: {
    userSelect: "none",
    padding: "0.3em 0.7em 0.2em",
    color: "white",
    textDecoration: "none",
    backgroundImage: "none",
    textShadow: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    borderRadius: "6px",
    textAlign: "center",
    verticalAlign: "center",
    transform: "translate(0, 0)",
    transition: "all .2s ease",
    cursor: "pointer",
    overflow: "hidden",
    boxShadow: "0px 6px 0px var(--boxShadowColor, #000)",
    "&:hover": {
      color: "white",
      transform: "translate(0, -2px)",
      boxShadow: "0px 8px 0px  var(--boxShadowColor, #000)",
      filter: "brightness(110%)",
    },
    "&:active": {
      transform: "translate(0, 3px)",
      boxShadow: "0px 2px 0px  var(--boxShadowColor, #000)",
    },
  },
};
