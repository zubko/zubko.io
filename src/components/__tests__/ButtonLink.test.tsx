import React from 'react';
import renderer from 'react-test-renderer';

import ButtonLink from '../ButtonLink';

describe('ButtonLink', () => {
  it('renders without regressions', () => {
    const tree = renderer
      .create(
        <ButtonLink
          backgroundColor="red"
          borderColor="blue"
          to="https://button.link.to">
          Title
        </ButtonLink>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
