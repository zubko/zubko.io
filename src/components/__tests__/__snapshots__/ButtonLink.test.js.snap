// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink renders without regressions 1`] = `
<a
  css={
    Array [
      Object {
        "&:active": Object {
          "boxShadow": "0px 2px 0px  var(--boxShadowColor, #000)",
          "transform": "translate(0, 3px)",
        },
        "&:hover": Object {
          "boxShadow": "0px 8px 0px  var(--boxShadowColor, #000)",
          "color": "white",
          "filter": "brightness(110%)",
          "transform": "translate(0, -2px)",
        },
        "alignItems": "center",
        "backgroundImage": "none",
        "borderRadius": "6px",
        "boxShadow": "0px 6px 0px var(--boxShadowColor, #000)",
        "color": "white",
        "cursor": "pointer",
        "display": "flex",
        "justifyContent": "center",
        "overflow": "hidden",
        "padding": "0.3em 0.7em 0.2em",
        "textAlign": "center",
        "textDecoration": "none",
        "textShadow": "none",
        "transform": "translate(0, 0)",
        "transition": "all .2s ease",
        "userSelect": "none",
        "verticalAlign": "center",
      },
      Object {
        "--boxShadowColor": "blue",
        "backgroundColor": "red",
      },
    ]
  }
  href="https://button.link.to"
  rel="noopener noreferrer"
  target="_blank"
>
  Title
</a>
`;
